napari/layers/base/base.py

```diff
@@ -524,15 +524,6 @@ def metadata(self, value: dict) -> None:
     def source(self):
         return self._source
 
-    @property
-    def loaded(self) -> bool:
```

Comment:
Even though this is only really relevant for the old async, it's also part of the napari public API and therefore should be deprecated instead of just removed.

My strong suggestion is to avoid that in this PR, which is more about removing the config integration. Could be done with some of the other simplifications though (i.e. when removing `ImageSliceData` and friends).

Reply:
Looks like there are also some related changes elsewhere, so those would need to change too. Happy to chat the general reasoning for the separation on Monday.

Reply:
This was related to an async trigger that I removed. I guess I just took this out along with it, but I can see how it should be deprecated instead. I can fix.